Components for FormA11y

PaliwalSparsh opened the pull request.

PaliwalSparsh committed:
Skiff_files: Add preprocessed form

PaliwalSparsh committed:
Add new page to ui

PaliwalSparsh committed:
Add homepage

PaliwalSparsh committed:
PDFPage: Add comments

PaliwalSparsh committed:
components: Add header component

PaliwalSparsh committed:
Create homepage

PaliwalSparsh committed:
Add steps

PaliwalSparsh committed:
refactor Steps

PaliwalSparsh committed:
Add icon button and dialog

PaliwalSparsh committed:
components: Add action Menu

PaliwalSparsh committed:
components: Fix action menu style

PaliwalSparsh committed:
add comments for better understanding

PaliwalSparsh committed:
Add explanation to codebase and fix size of PDF element

PaliwalSparsh committed:
Remove logs

PaliwalSparsh committed:
DuplicateModal: Fix styles

PaliwalSparsh committed:
components: Add zoom component

PaliwalSparsh committed:
Add comments

PaliwalSparsh committed:
feature: Restrict overlapping of fields by stopping propogation

PaliwalSparsh committed:
feature: show currently selected box

PaliwalSparsh committed:
feature: Add newly created field to selection. Clicking outside created field removes selection.

PaliwalSparsh committed:
feature: Update selection style

PaliwalSparsh committed:
feature: Restrict creation of annotation with small bounds

PaliwalSparsh committed:
Show action menu for selected annotation

PaliwalSparsh committed:
Turn off user text select, prevents text selection while selecting multiple fields

PaliwalSparsh committed:
features: Show action menu for just one selection; Delete multiple annotations

joshuaharry commented on README.md:
````diff
@@ -1,3 +1,14 @@
+
+Preprocessing PDF
+```
+pawls add /Users/sparshpaliwal/Desktop/repos/pawls/skiff_files/apps/pawls/papers/sample_form.pdf
````
Could we potentially include a link to the `sample_form.pdf` file so that others can access it more easily? I don't have it myself, so an easy way of downloading it would be great.

PaliwalSparsh replied:
Great suggestion! Added the file.

joshuaharry commented on ui/src/App.tsx:
```diff
@@ -57,6 +57,7 @@ const App = () => {
             <BrowserRouter>
                 <Route path="/" exact component={RedirectToFirstPaper} />
                 <Route path="/pdf/:sha" component={PDFPage} />
+                <Route path="/home/:sha" component={HomePage} />
```
I appreciate the strategy of keeping the old code around! This seems like a really great way of making sure we can keep accessing the old logic; nice.

PaliwalSparsh replied:
Great that we are on the same page. As we get more confident with the codebase, we can slowly start removing the code and making it leaner.

joshuaharry commented on ui/src/components/formA11y/Header.tsx:
```diff
@@ -0,0 +1,61 @@
+    );
+};
+
+const stepInfo = [
```
Is `stepInfo` ever meant to change while we're running the application?

PaliwalSparsh replied:
These are the sub-steps for remediating the form. They will stay the same. We can move the array to another place as well. Feel free to update any of the components as needed.

joshuaharry commented on ui/src/context/AnnotationStore.ts:
```diff
@@ -97,6 +99,7 @@ export class PdfAnnotations {
     }
 
     withNewAnnotation(a: Annotation): PdfAnnotations {
+        console.log('Logging newly created annotation:', a);
```
Do we always want to log this information? I find this very useful for debugging, but I'm not sure whether we want to commit this directly into the repository—perhaps we could add a line about this trick in the README?

joshuaharry replied:
What do you think about adding a `deleteAnnotations` method to the class?

PaliwalSparsh replied:
> What do you think about adding a `deleteAnnotations` method to the class?

Luckily, we already have a `deleteAnnotation`. However, there are no class methods for performing bulk operations, so we work on those. 

For example in this commit -
https://github.com/PaliwalSparsh/Form-A11y/pull/1/commits/16cdbf15937e0605928700daf80c954f190fb8f8

I had to write code for bulk deletion in the Selection component. Clearly, it could have been a part of the PDFAnnotations class.

PaliwalSparsh replied:
> Do we always want to log this information? I find this very useful for debugging, but I'm not sure whether we want to commit this directly into the repository—perhaps we could add a line about this trick in the README?

Oops! Something I did by mistake. No pre-commit hook to save me :P

I will remove this. However, this log is super useful for development.

joshuaharry reviewed (approved):
Overall, LGTM!

joshuaharry commented:
LGTM!

PaliwalSparsh committed:
chore: add sample form

PaliwalSparsh committed:
Solves #4: Create tool modes
Add ToolStore context
Create tool specific properties
Make tool panel functional

joshuaharry commented on ui/src/context/ToolStore.ts:
```diff
@@ -0,0 +1,15 @@
+    MOVE: 'MOVE',
+};
+
+export const ToolStore = createContext({
```
Nice! This fits with the style of the existing code.

joshuaharry commented on ui/src/pages/HomePage.tsx:
```diff
@@ -214,53 +217,55 @@ export const HomePage = () => {
-                            onError,
-                        }}>
-                        <AnnotationStore.Provider
+                    <ToolStore.Provider value={{ currentTool, setCurrentTool }}>
```
Great job! This is how to use context.

joshuaharry reviewed (approved):
LGTM!
